test/passes/asyncify.wast

```diff
@@ -0,0 +1,53 @@
+(module
+  (type $f (func (param i32)))
+  (memory 1 2)
+  (import "env" "import" (func $import))
+  (import "env" "import2" (func $import2 (param i32)))
+  (table 1 1)
+  (func $liveness1 (param $live0 i32) (param $dead0 i32)
+    (local $live1 i32)
+    (local $dead1 i32)
+    (drop (local.get $dead0))
+    (drop (local.get $dead1))
+    (call $import)
+    (drop (local.get $live0))
+    (drop (local.get $live1))
+  )
+  (func $liveness2 (param $live0 i32) (param $dead0 i32)
+    (local $live1 i32)
+    (local $dead1 i32)
+    (drop (local.get $dead0))
+    (drop (local.get $dead1))
+    (call $import)
+    (drop (local.get $live0))
+    (drop (local.get $live1))
+  )
+  (func $liveness3 (param $live0 i32) (param $dead0 i32)
+    (local $live1 i32)
+    (local $dead1 i32)
+    (call $import)
+    (drop (local.get $live0))
+    (call $import)
+    (drop (local.get $live1))
+  )
+  (func $liveness4 (param $live0 i32) (param $dead0 i32)
+    (if (i32.const 0)
+      (call $import)
+    )
+    (drop (local.get $live0))
```

Comment:
Why is this local considered live if it's only dropped? AFAIK `local.get` doesn't have any side-effects on its own?

Reply:
The liveness analysis doesn't check if a use is silly or not, it just looks for uses. So even a drop of a get is something that is a use.

This is just a separation of concerns - other passes (vacuum, coalesce-locals) would remove such a silly use. So it's only noticeable in an artificial testcase like this, where we run asyncify without any other optimizations.

Reply:
Ah, good to know.